tavern/core.py

```diff
@@ -293,7 +289,28 @@ def _run_pytest(
     if tavern_strict:
         pytest_args += ["--tavern-strict", tavern_strict]
 
-    return pytest.main(args=pytest_args)
+    with ExitStack() as stack:
+        if tavern_global_cfg:
```

Comment:
Might be worth breaking out this block to a method so it can have a name like `get_config_from_file_or_dict()`. Maybe I've been using django too much lately ;)

Reply:
Done